plugin/src/main/scala/chisel3/internal/plugin/ChiselComponent.scala

```diff
@@ -89,7 +89,8 @@ class ChiselComponent(val global: Global, arguments: ChiselPluginArguments)
       shouldMatchGen(
         tq"chisel3.BaseType",
         tq"chisel3.MemBase[_]",
-        tq"chisel3.VerificationStatement"
+        tq"chisel3.VerificationStatement",
+        tq"chisel3.properties.DynamicObject"
```

Comment:
A++, I was getting ready to point out that this is needed but you already figured it out 🙂